source/language/types.rst

```diff
@@ -119,13 +119,24 @@ the quantum memory.
 Physical Qubits
 ~~~~~~~~~~~~~~~
 
-While program qubits can be named, hardware qubits are referenced only
+During quantum compilation it is critical to understand if an operation is being applied
+to an "virtual qubit" (a qubit pre-assignment) or a "physical" hardware qubit
+within a device. In most hardware implementations no two physical qubits are alike
```

Comment:
I think "a qubit pre-assignment" is slightly dangerous terminology, and might belie something important to discuss when we're talking about virtual/physical qubits.

This is how I (as a high-level quantum compiler writer) see the split: A virtual qubit as defined by a high-level user is not necessarily represented by the same hardware qubit throughout an entire circuit, because the compiler (a high-level quantum-aware component of it - Qiskit, for example) may need to do a bunch of swap-mapping to satisfy backend connectivity constraints.  By the time we hit the QSS stack (in IBM land), we're assuming that this is already completely resolved, and so with _that_ hat on, the view "`qubit q;` causes `q` to be associated with an unknown hardware qubit" would be ok, but (I think) only really because in that situation there'd be no reason to ever write `qubit q;` - you'd already know your hardware qubits.  At the high-level of an OQ3 program that's before swap mapping has been performed, we can't make that direct association between one virtual and one physical, and that's where I get really nervous about allowing _mixed_ declarations.

If that's not the same as how others see the split, perhaps it's something we ought to clarify - I'm not sure we can entirely make sense of the difference between virtual/hardware in the spec until we're all on the same page.

(Also: 'an "virtual qubit"' -> 'a "virtual qubit"'.)